src/libraries/System.Private.CoreLib/src/System/IO/Stream.cs

```diff
@@ -332,6 +332,48 @@ static async ValueTask<int> FinishReadAsync(Task<int> readTask, byte[] localBuff
             }
         }
 
+        public async ValueTask ReadExactlyAsync(Memory<byte> buffer, CancellationToken cancellationToken = default) =>
+            _ = await ReadAtLeastAsyncCore(buffer, buffer.Length, throwOnEndOfStream: true, cancellationToken).ConfigureAwait(false);
+
+        public async ValueTask ReadExactlyAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken = default)
+        {
+            ValidateBufferArguments(buffer, offset, count);
```

Comment:
These argument exceptions are going to be stored into the returned task rather than propagated.  We generally try to avoid that for ArgumentXxException.